rust/connlib/libs/tunnel/src/lib.rs

```diff
@@ -224,7 +224,17 @@ where
                 }
             }
         }
-        self.resources.write().insert(resource_description);
+        let resource_list = {
+            let mut resources = self.resources.write();
+            resources.insert(resource_description);
+            resources.resource_list()
+        };
+        match resource_list {
+            Ok(resource_list) => {
+                self.callbacks.on_update_resources(resource_list);
```

Comment:
This is fine now but when we hook up the tunnel stack this should be in `iface_config.add_route`, since this is how the clients will update the routing table.

Reply:
Noted; I'll just address this now so we can't forget.

Reply:
Update: doing that involves some restructuring, so I'll just do that for my next PR and keep things moving.